distutils/dist.py

```diff
@@ -829,7 +835,11 @@ def get_command_class(self, command):
 
         raise DistutilsModuleError(f"invalid command '{command}'")
 
-    def get_command_obj(self, command, create=True):
+    @overload
+    def get_command_obj(self, command: str, create: Literal[True] = True) -> Command: ...
+    @overload
+    def get_command_obj(self, command: str, create: Literal[False]) -> Command | None: ...
+    def get_command_obj(self, command: str, create: bool = True) -> Command | None:
```

Comment:
This complicated signature, caused by a boolean parameter, should probably be re-written as two methods, `get_command_obj` and `get_cached_command_obj`, where the latter just returns `self.command_obj.get(command)`. Of course, that's not a concern for this PR.